cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,91 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadTestUtilsBundle struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func apiProviderDownloadMockUtilsBundle() *apiProviderDownloadTestUtilsBundle {
+	utilsBundle := apiProviderDownloadTestUtilsBundle{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return &utilsBundle
+}
+
+//Successful API Provider download cases
+func TestApiProviderDownloadSuccess(t *testing.T) {
+	t.Parallel()
+	t.Run("Successful Download of API Provider", func(t *testing.T) {
+		file, err := ioutil.TempFile("", "provider1.json")
+		if err != nil {
+			t.FailNow()
```

Comment:
There is also the require.NoError function for this. From the documentaion:
> Package require implements the same assertions as the `assert` package but stops test execution when a test fails.

Reply:
Not required anymore, as it is already mocked and assert.NoError already implemented there. Refer line #45